### PR TITLE
Issue #121 Norway Wheelchair equipment fix3 - rebased on Master

### PR DESCRIPTION
**Pull request by nick-knowles**
Errors in VehicleType / ActualVehicleEquipment, WheelchairVehicleEquipment  , AccessVehicleEquipment
  * WWheelchairEquipmentRef renamed to WheelchairVehicleEquipmentRef
  * Supertypes corrected

### CODE REVIEW

**Comment by skinkie** [comment]
@Aurige @ue71603  this needs a complete review.

The WheelchairVehicleEquipmentRef is already in our NeTEx_publication.xsd but not in our actual files. This pull request changes a lot of things, must be **validated**.

**Comment by skinkie** [comment]
@Aurige @ue71603  this needs a complete review.

The WheelchairVehicleEquipmentRef is already in our NeTEx_publication.xsd but not in our actual files. This pull request changes a lot of things, must be **validated**.

File: NeTEx.spp [hunk 1]
```diff
@@ -577,18 +577,7 @@
 						<File FilePath="examples\functions\fares\Netex_91.1_Rail_RailCard_MultipleProduct.xml" HomeFolder="Yes"/>
 					</Folder>
 					<Folder FolderName="bus">
-						<Folder FolderName="uk">
```
Comment (ue71603) [inline NeTEx.spp:580]: I suggest not to update NeTEx.spp

File: README.md [hunk 2]
```diff
@@ -3,16 +3,8 @@
```
Comment (ue71603) [inline README.md:3]: I suggest not to update README.md from such an old PR

File: xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_support.xsd [hunk 3]
```diff
@@ -53,6 +60,19 @@ Rail transport, Roads and Road transport
 	</xsd:annotation>
 	<!-- ======================================================================= -->
 	<!-- ======================================================================= -->
+	<xsd:complexType name="vehicleEquipments_RelStructure">
```
Comment (ue71603) [inline xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_support.xsd:63]: Already exists and even in an extended form:
	<xsd:complexType name="vehicleEquipments_RelStructure">
		<xsd:annotation>
			<xsd:documentation>List of VEHICLE EQUIPMENT.</xsd:documentation>
		</xsd:annotation>
		<xsd:complexContent>
			<xsd:extension base="containmentAggregationStructure">
				<xsd:choice maxOccurs="unbounded">
					<xsd:element ref="AccessVehicleEquipment"/>
					<xsd:element ref="WheelchairVehicleEquipment"/>
				</xsd:choice>
			</xsd:extension>
		</xsd:complexContent>
	</xsd:complexType>
	<!-- ====ACCESS=====

File: xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_support.xsd [hunk 4]
```diff
@@ -61,12 +81,12 @@ Rail transport, Roads and Road transport
 	</xsd:simpleType>
 	<xsd:element name="VehicleEquipmentRef" type="InstalledEquipmentRefStructure" substitutionGroup="PassengerEquipmentRef">
 		<xsd:annotation>
-			<xsd:documentation>Reference to a WHEELCHAIR VEHICLE EQUIPMENT.</xsd:documentation>
```
Comment (ue71603) [inline xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_support.xsd:64]: should stay in (if git can find it)

File: xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_support.xsd [hunk 5]
```diff
@@ -61,12 +81,12 @@ Rail transport, Roads and Road transport
...
+			<xsd:documentation>Reference to a  VEHICLE EQUIPMENT.</xsd:documentation>
 		</xsd:annotation>
 	</xsd:element>
 	<xsd:complexType name="VehicleEquipmentRefStructure" abstract="true">
 		<xsd:annotation>
-			<xsd:documentation>Type for a reference to a WHEELCHAIR VEHICLE EQUIPMENT.</xsd:documentation>
```
Comment (ue71603) [inline xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_support.xsd:69]: should stay in

File: xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_support.xsd [hunk 6]
```diff
@@ -111,12 +131,17 @@ Rail transport, Roads and Road transport
 		</xsd:annotation>
 		<xsd:restriction base="VehicleEquipmentIdType"/>
 	</xsd:simpleType>
-	<xsd:element name="WheelchairVehicleRef" type="InstalledEquipmentRefStructure" substitutionGroup="VehicleEquipmentRef">
```
Comment (ue71603) [inline xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_support.xsd:114]: Is already done.

File: xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_support.xsd [hunk 7]
```diff
@@ -111,12 +131,17 @@ Rail transport, Roads and Road transport
...
+	<xsd:element name="WheelchairVehicleEquipmentRef" type="WheelchairVehicleEquipmentRefStructure" substitutionGroup="VehicleEquipmentRef">
 		<xsd:annotation>
 			<xsd:documentation>Reference to a WHEELCHAIR VEHICLE EQUIPMENT.</xsd:documentation>
 		</xsd:annotation>
 	</xsd:element>
-	<xsd:complexType name="WheelchairVehicleRefStructure" abstract="false">
```
Comment (ue71603) [inline xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_support.xsd:119]: It is in master, but differently (different type). Needs to be discussed.
	<xsd:element name="WheelchairVehicleRef" type="InstalledEquipmentRefStructure" substitutionGroup="VehicleEquipmentRef">
		<xsd:annotation>
			<xsd:documentation>Reference to a WHEELCHAIR VEHICLE EQUIPMENT.</xsd:documentation>
		</xsd:annotation>
	</xsd:element>

File: xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_version.xsd [hunk 8]
```diff
@@ -62,63 +67,58 @@ Rail transport, Roads and Road transport
 	<xsd:include schemaLocation="../netex_genericFramework/netex_accessibility/netex_acsb_accessibility.xsd"/>
 	<xsd:include schemaLocation="netex_equipmentVehiclePassenger_support.xsd"/>
 	<xsd:include schemaLocation="netex_vehicleType_support.xsd"/>
-	<xsd:group name="VehicleEquipmentChoice">
```
Comment (ue71603) [inline xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_version.xsd:65]: still there in master, but used nowhere. So we could remove it.

File: xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_version.xsd [hunk 9]
```diff
@@ -62,63 +67,58 @@ Rail transport, Roads and Road transport
...
-			<xsd:extension base="containmentAggregationStructure">
-				<xsd:choice maxOccurs="unbounded">
-					<xsd:element ref="AccessVehicleEquipment"/>
-					<xsd:element ref="WheelchairVehicleEquipment"/>
-				</xsd:choice>
-			</xsd:extension>
-		</xsd:complexContent>
-	</xsd:complexType>
 	<!-- ====ACCESS=================================================================== -->
-	<xsd:element name="ActualVehicleEquipment" type="ActualVehicleEquipment_VersionStructure" abstract="true" substitutionGroup="PassengerEquipment">
```
Comment (ue71603) [inline xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_version.xsd:88]: still in master

File: xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_version.xsd [hunk 10]
```diff
@@ -62,63 +67,58 @@ Rail transport, Roads and Road transport
...
+	<!-- =========== ACCESS VEHICLE EQUIPMENT =========================================== -->
+	<xsd:element name="VehicleEquipment" substitutionGroup="PassengerEquipment">
```
Comment (ue71603) [inline xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_version.xsd:72]: new to master. may have impact on other things

File: xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_version.xsd [hunk 11]
```diff
@@ -62,63 +67,58 @@ Rail transport, Roads and Road transport
...
 		<xsd:annotation>
-			<xsd:documentation>An item of EQUIPMENT of a particular type actually available in an individual VEHICLE.</xsd:documentation>
+			<xsd:documentation>Specialisation of VEHICLE EQUIPMENT for ACCESS providing information such as low floor, ramp, access area dimensions, etc.</xsd:documentation>
```
Comment (ue71603) [inline xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_version.xsd:74]: shouldn't it be AccessVehicleEquipment in that case?

File: xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_version.xsd [hunk 12]
```diff
@@ -176,19 +176,9 @@ Rail transport, Roads and Road transport
 			<xsd:group ref="AccessVehicleEquipmentAdditionalGroup"/>
 		</xsd:sequence>
 	</xsd:group>
-	<xsd:group name="AccessVehicleEquipmentGroup2">
```
Comment (ue71603) [inline xsd/netex_framework/netex_reusableComponents/netex_equipmentVehiclePassenger_version.xsd:179]: removing this makes sense

File: xsd/netex_framework/netex_reusableComponents/netex_vehicleType_support.xsd [hunk 13]
```diff
@@ -321,6 +355,15 @@ Rail transport, Roads and Road transport
 			<xsd:enumeration value="naturalGas"/>
 			<xsd:enumeration value="biodiesel"/>
 			<xsd:enumeration value="electricity"/>
+			<xsd:enumeration value="electricContact"/>
```
Comment (ue71603) [inline xsd/netex_framework/netex_reusableComponents/netex_vehicleType_support.xsd:358]: changing things here may break the whole thing.

**Comment by ue71603** [comment]
I think @Aurige needs to check as well. Some parts need to be removed from the PR and the rest could then be submitted.

File: xsd/netex_framework/netex_reusableComponents/netex_vehicleType_version.xsd [hunk 14]
```diff
@@ -777,6 +783,80 @@ Rail transport, Roads and Road transport
...
+		</xsd:annotation>
+		<xsd:complexContent>
+			<xsd:extension base="Equipment_VersionStructure">
+				<xsd:sequence>
+					<xsd:group ref="ActualVehicleEquipmentGroup"/>
+				</xsd:sequence>
+			</xsd:extension>
+		</xsd:complexContent>
+	</xsd:complexType>
+	<xsd:group name="ActualVehicleEquipmentGroup">
```
Comment (skinkie) [inline xsd/netex_framework/netex_reusableComponents/netex_vehicleType_version.xsd:841]: This is also different in master.

**Comment by Aurige** [comment]
It looks that we already have what we need ... and there are breaking changes here...  I would rather forget this PR
I will add it to the discussion on Monday to make sure.

File: xsd/netex_framework/netex_reusableComponents/netex_vehicleType_support.xsd [hunk 15]
```diff
@@ -285,6 +292,33 @@ Rail transport, Roads and Road transport
...
+	</xsd:element>
+	<xsd:complexType name="ActualVehicleEquipmentRefStructure">
+		<xsd:annotation>
+			<xsd:documentation>Type for a reference to a ACTUAL VEHICLE EQUIOMENT.</xsd:documentation>
+		</xsd:annotation>
+		<xsd:simpleContent>
+			<xsd:restriction base="VersionOfObjectRefStructure">
+				<xsd:attribute name="ref" type="ActualVehicleEquipmentIdType" use="required">
+					<xsd:annotation>
+						<xsd:documentation>Identifier of a ACTUAL VEHICLE EQUIOMENT.</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_framework/netex_reusableComponents/netex_vehicleType_support.xsd:315]:
```suggestion
						<xsd:documentation>Identifier of a ACTUAL VEHICLE EQUIPMENT.</xsd:documentation>
```

File: xsd/netex_framework/netex_reusableComponents/netex_vehicleType_support.xsd [hunk 16]
```diff
@@ -285,6 +292,33 @@ Rail transport, Roads and Road transport
...
+	<!-- ==== ACTUAL VEHICLE EQUIPMENT================================================ -->
+	<xsd:simpleType name="ActualVehicleEquipmentIdType">
+		<xsd:annotation>
+			<xsd:documentation>Type for identifier of a ACTUAL VEHICLE EQUIOMENT.</xsd:documentation>
+		</xsd:annotation>
+		<xsd:restriction base="ObjectIdType"/>
+	</xsd:simpleType>
+	<xsd:element name="ActualVehicleEquipmentRef" type="ActualVehicleEquipmentRefStructure" substitutionGroup="VersionOfObjectRef">
+		<xsd:annotation>
+			<xsd:documentation>Reference to a ACTUAL VEHICLE EQUIOMENT.</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_framework/netex_reusableComponents/netex_vehicleType_support.xsd:304]:
```suggestion
			<xsd:documentation>Reference to a ACTUAL VEHICLE EQUIPMENT.</xsd:documentation>
```

File: xsd/netex_framework/netex_reusableComponents/netex_vehicleType_support.xsd [hunk 17]
```diff
@@ -285,6 +292,33 @@ Rail transport, Roads and Road transport
 		</xsd:simpleContent>
 	</xsd:complexType>
 	<!-- ======================================================================= -->
+	<!-- ==== ACTUAL VEHICLE EQUIPMENT================================================ -->
+	<xsd:simpleType name="ActualVehicleEquipmentIdType">
+		<xsd:annotation>
+			<xsd:documentation>Type for identifier of a ACTUAL VEHICLE EQUIOMENT.</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_framework/netex_reusableComponents/netex_vehicleType_support.xsd:298]:
```suggestion
			<xsd:documentation>Type for identifier of an ACTUAL VEHICLE EQUIPMENT.</xsd:documentation>
```

**Comment by Aurige** [comment]
As said before, it's an old PR that is already mostly covered... I would close this one 

The only good thing looks to be the 			
`<xsd:element name="actualVehicleEquipments" type="actualVehicleEquipments_RelStructure" minOccurs="0">`
instead of 
`<xsd:element name="actualVehicleEquipments" type="equipments_RelStructure" minOccurs="0">`

But since an actualVehicleEquipment is an equipment that's just adding a constraint, not adding a new feature. We may to a separated PR for this.

**Pull request closed without merging**